File: 0884-uncommon-words-from-two-sentences/0884-uncommon-words-from-two-sentences.java
class Solution {
    public String[] uncommonFromSentences(String s1, String s2) {
        Map<String, Integer> wordDict = new HashMap<>();
        List<String> res = new ArrayList<>();

        for (String word: s1.split(" ")) {
            wordDict.put(word, wordDict.getOrDefault(word, 0) + 1);
        }

        for (String word: s2.split(" ")) {
            wordDict.put(word, wordDict.getOrDefault(word, 0) + 1);
        }

        for (String word: wordDict.keySet()) {
            if (wordDict.get(word) == 1) {
                res.add(word);
            }
        }

        return res.toArray(new String[res.size()]);
    }
}

File: 0884-uncommon-words-from-two-sentences/0884-uncommon-words-from-two-sentences.py
class Solution:
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:
        wordDict = defaultdict(int)

        for word in s1.split():
            wordDict[word] += 1

        for word in s2.split():
            wordDict[word] += 1

        return [word for word in wordDict.keys() if wordDict[word] == 1]